### PR TITLE
Treat non-dict JSON function arguments as empty (instead of unexpected raise)

### PR DESCRIPTION
**Pull request by GICodeWarrior**
# Description

Currently, when an OpenAI LLM returns invalid JSON as function call arguments, the arguments are treated as empty.

This change extends that behavior to valid JSON which is not a dictionary (instead of raising unexpectedly).

See here for the unexpected exception this PR fixes:
https://github.com/run-llama/llama_index/actions/runs/11128743328/job/30924194701
```python
            # this should handle both complete and partial jsons
            try:
                argument_dict = parse_partial_json(tool_call.function.arguments)
            except ValueError:
                argument_dict = {}
    
            tool_selections.append(
>               ToolSelection(
                    tool_id=tool_call.id,
                    tool_name=tool_call.function.name,
                    tool_kwargs=argument_dict,
                )
            )
E           pydantic_core._pydantic_core.ValidationError: 1 validation error for ToolSelection
E           tool_kwargs
E             Input should be a valid dictionary [type=dict_type, input_value=None, input_type=NoneType]
E               For further information visit https://errors.pydantic.dev/2.9/v/dict_type

llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py:950: ValidationError
```

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] Added new unit/integration tests
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py [hunk 1]
```diff
@@ -946,6 +946,10 @@ def get_tool_calls_from_response(
             except ValueError:
                 argument_dict = {}
 
+            # Treat parsable non-dict JSON as empty
+            if not isinstance(argument_dict, dict):
+                argument_dict = {}
+
             tool_selections.append(
                 ToolSelection(
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py:954]: Couldn't we just update the `ToolSelection` object to default to an empty dict? Then it generalizes to every LLM

Comment (GICodeWarrior) [reply llama-index-integrations/llms/llama-index-llms-openai/llama_index/llms/openai/base.py:954]: OK, I'll make that change.

**Comment by GICodeWarrior** [comment]
@logan-markewich any tips regarding this test failure?  It seems unrelated to this PR.
```python
_ ERROR collecting llama-index-integrations/selectors/llama-index-selectors-notdiamond/tests/test_selectors_notdiamond.py _
ImportError while importing test module 'llama-index-integrations/selectors/llama-index-selectors-notdiamond/tests/test_selectors_notdiamond.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:1[26](https://github.com/run-llama/llama_index/actions/runs/11129534683/job/30926844526?pr=16316#step:9:27): in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
llama-index-integrations/selectors/llama-index-selectors-notdiamond/tests/test_selectors_notdiamond.py:13: in <module>
    from llama_index.selectors.notdiamond.base import NotDiamondSelector, LLMSingleSelector
llama-index-integrations/selectors/llama-index-selectors-notdiamond/llama_index/selectors/notdiamond/__init__.py:1: in <module>
    from llama_index.selectors.notdiamond.base import NotDiamondSelector
llama-index-integrations/selectors/llama-index-selectors-notdiamond/llama_index/selectors/notdiamond/base.py:12: in <module>
    from notdiamond import NotDiamond, LLMConfig, Metric
/home/runner/.cache/pants/named_caches/pex_root/venvs/s/2ff4cd5a/venv/lib/python3.10/site-packages/notdiamond/__init__.py:1: in <module>
    from notdiamond.llms.client import NotDiamond  # noqa: F401
/home/runner/.cache/pants/named_caches/pex_root/venvs/s/2ff4cd5a/venv/lib/python3.10/site-packages/notdiamond/llms/client.py:22: in <module>
    from litellm import token_counter
E   ModuleNotFoundError: No module named 'litellm'
```

**Comment by GICodeWarrior** [comment]
I'm not sure how to fix the build issues here.

`llama-index-integrations/selectors/llama-index-selectors-notdiamond` is complaining that litellm isn't installed.  It looks like a recent version of the notdiamond library removed litellm as a requirement and then the latest version added it back in.  I tried excluding the affected version, but that didn't seem to help.

There is also this error:
```
ERROR: Could not find a version that satisfies the requirement tablestore<7.0.0,>=6.0.0 (from versions: 4.0.0, 4.1.0, 4.2.0, 4.3.1, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.5.2, 4.6.0, 5.0.0, 5.1.0, 5.2.0, 5.2.1, 5.3.0, 5.4.0, 5.4.1, 5.4.2, 5.4.3, 5.4.4, 5.5.0, 5.5.1, 6.0.0)
```

It's not clear to me where that is triggered, but I only see tablestore mentioned in `llama-index-integrations/vector_stores/llama-index-vector-stores-tablestore`.  Also, the requirement `tablestore<7.0.0,>=6.0.0` includes `6.0.0`, which it explicitly claims does not satisfy the requirement...

Maybe one error is causing the other?

**Comment by Georgehe4** [comment]
Same issue in a few other tests as well, looks like an unrelated breakage #16309

**Comment by GICodeWarrior** [comment]
Unrelated redis connection error now. :-(

**Comment by logan-markewich** [comment]
Seems like the redis test is flakey somehow, safe to ignore

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**